--- ./custom.dts
+++ ./custom.dts
@@ -6,8 +6,7 @@
   compatible = "ucbbar,spike-bare-dev";
   model = "ucbbar,spike-bare";
   chosen {
-    stdout-path = &SERIAL0;
-    bootargs = "console=ttyS0 earlycon";
+    bootargs = "console=hvc0 earlycon=sbi nokaslr";
   };
   cpus {
     #address-cells = <1>;
@@ -35,6 +34,21 @@
     device_type = "memory";
     reg = <0x0 0x80000000 0x0 0x80000000>;
   };
+  reserved-memory {
+          #address-cells = <0x02>;
+          #size-cells = <0x02>;
+          ranges;
+
+          mmode_resv1@80020000 {
+                  reg = <0x00 0x80020000 0x00 0x20000>;
+                  no-map;
+          };
+
+          mmode_resv0@80000000 {
+                  reg = <0x00 0x80000000 0x00 0x20000>;
+                  no-map;
+          };
+  };
   soc {
     #address-cells = <2>;
     #size-cells = <2>;
@@ -78,6 +78,12 @@
       reg-shift = <0x0>;
       reg-io-width = <0x1>;
     };
+    virtio9p: virtio@40011000 {
+      compatible = "virtio,mmio";
+      interrupt-parent = <&PLIC>;
+      interrupts = <2>;
+      reg = <0x0 0x40011000 0x0 0x1000>;
+    };
   };
   htif {
     compatible = "ucb,htif0";
